core/src/main/java/org/bitcoinj/core/StoredBlock.java

```diff
@@ -142,6 +144,33 @@ public static StoredBlock deserializeCompact(NetworkParameters params, ByteBuffe
         return new StoredBlock(params.getDefaultSerializer().makeBlock(header), chainWork, height);
     }
 
+    /** Serializes the stored block to a custom packed format. Used by {@link org.bitcoinj.store.SPVBlockStore}. */
```

Comment:
I think it's a bit unfortunate that these two methods duplicate the above two methods. In theory, we could implement the new methods like this:

```
FileChannel channel = randomAccessFile.getChannel();
ByteBuffer buffer = ByteBuffer.allocate(CHAIN_WORK_BYTES);
deserializeCompact(params, buffer);
```

Of course I don't about the performance implications and wether we run into specific problems in Windows again. Have you considered this?

Reply:
I added a commit with your suggestion "Avoid duplicating StoredBlock's serializeCompact() and deserializeCompact".
I will squash once we finish the discussion and testing.
I don't think this deduplication commit brings up the windows problem again, since the windows problem is just with mmaped files not with the uses of Buffers or FileChannels.
About performance, this deduplication commit does not improve/deteriorate performance. I will address your performance question in a broader PR comment.